pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1102,6 +1102,16 @@ def test_merge_incompat_dtypes_error(self, df1_vals, df2_vals):
         with pytest.raises(ValueError, match=msg):
             pd.merge(df2, df1, on=['A'])
 
+    def test_merge_on_index_with_more_values(self):  # GH 24212
+        df1 = pd.DataFrame([[1, 2], [2, 4], [3, 6], [4, 8]],
+                           columns=['a', 'b'])
+        df2 = pd.DataFrame([[3, 30], [4, 40]],
+                           columns=['a', 'c'])
+        df1.set_index('a', drop=False, inplace=True)
+        df2.set_index('a', inplace=True)
+        result = pd.merge(df1, df2, left_index=True, right_on='a', how='left')
+        assert 1 in result.index
```

Comment:
Hmm...I think we should just check the entire `DataFrame` instead of just an element in `result.index`.  Also, comment on what's being tested for here.